majordodo-core/src/main/java/majordodo/replication/ReplicatedCommitLog.java

```diff
@@ -1042,12 +1074,24 @@ public void followTheLeader(LogSequenceNumber skipPast, BiConsumer<LogSequenceNu
                     lastSequenceNumber = number.sequenceNumber;
                     currentLedgerId = number.ledgerId;
                 }
+            } catch (BKException err) {
```

Comment:
it is better to leave off all of these refactor of exception handling
it is very dangerous and it makes the review more difficult